### PR TITLE
[v4] Real date selection

### PR DESCRIPTION
**Pull request by devinmatte**
Uses the real date from the selector and the url bar. Allows deep linking to a specific date

### CODE REVIEW

File: common/api/datadashboard.ts [hunk 1]
```diff
@@ -108,6 +108,7 @@ export const useCustomQueries: UseQueriesOverload = (
             ? fetchAggregateData(name, queries[name].params)
             : fetchSingleDayData(name, queries[name].params),
         enabled,
+        staleTime: 30000, // Don't refetch within 30 seconds
```
Comment (devinmatte) [inline common/api/datadashboard.ts:111]: This prevents spam to the API. We can likely bump this up to 1 minute with minimal to no user impact, but I decided to start out cautious

File: common/utils/date.ts [hunk 2]
```diff
@@ -25,3 +25,11 @@ export const getCurrentDate = (): string => {
   const maxDate = localDate.toISOString().split('T')[0];
   return maxDate;
 };
+
+export const getOffsetDate = (date: string): string => {
+  const isoDate = new Date(date);
+  const offset = isoDate.getTimezoneOffset();
+  const localDate = new Date(isoDate.valueOf() + offset * 60 * 1000);
+  const offsetDate = localDate.toISOString().split('T')[0];
+  return offsetDate;
```
Comment (skaplan-dev) [inline common/utils/date.ts:34]: we should be able to use `dayjs` utils here

File: modules/traveltimes/TravelTimesWidget.tsx [hunk 3]
```diff
@@ -31,14 +31,12 @@ export const TravelTimesWidget: React.FC = () => {
       [SingleDayAPIParams.fromStop]: fromStopIds || '',
       [SingleDayAPIParams.toStop]: toStopIds || '',
       [SingleDayAPIParams.stop]: fromStopIds || '',
-      [SingleDayAPIParams.date]: startDate,
+      [SingleDayAPIParams.date]: startDate ?? getCurrentDate(),
```
Comment (skaplan-dev) [inline modules/traveltimes/TravelTimesWidget.tsx:34]: could we set this default date in the `hook` itself? we're repeating it a lot

**Review by skaplan-dev — APPROVED** [review]
just few nits

**Pull request merged**